core/state/statedb.go

```diff
@@ -291,8 +290,11 @@ func (self *StateDB) CreateAccount(addr common.Address) vm.Account {
 // Setting, copying of the state methods
 //
 
-func (self *StateDB) Copy() *StateDB {
-	state := New(common.Hash{}, self.db)
+func (self *StateDB) Copy() (*StateDB, error) {
+	state, err := New(common.Hash{}, self.db)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Most of the other edits are cause by this change. Copy doesn't need to return an error because 
the state that is being copied already exists. The implementation of Copy uses New because it initializes
all the struct fields.